src/containers/views/Configuration.js

```diff
@@ -10,6 +11,17 @@ import { putConfig, onEditorChange } from '../../actions/config';
 
 export class Configuration extends Component {
 
+  componentDidMount() {
+    const { router, route } = this.props;
+    router.setRouteLeaveHook(route, this.routerWillLeave.bind(this));
+  }
+
+  routerWillLeave(nextLocation) {
+    const { editorChanged } = this.props;
+    if (editorChanged)
+      return 'Your work is not saved! Are you sure you want to leave?';
```

Comment:
somehow `Your work is not saved!` doesn't sound (grammatically) correct to me.. may i suggest:
**`Your work has not been saved!`** ? or **`You've unsaved changes on this page.`** ?